muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application/current-student/main-chart.tsx

```diff
@@ -148,41 +165,70 @@ class CurrentStudentStatistics extends React.Component<CurrentStudentStatisticsP
     }
     //NOTE: The unused data can be cut here. (Option 1)
     //NOTE: For the sake of keeping the same chart borders it might be wise to leave the data rows with 0 values, but keep date points.
-    let chartDataMap = new Map<string, {logins?: number, assignments?: number, exercises?: number}>();
-    chartDataMap.set(new Date().toISOString().slice(0, 10), {"logins": 0, "assignments": 0, "exercises": 0});
-      this.props.logins.map((login)=>{
-        let date = login.slice(0, 10);
-        let entry = chartDataMap.get(date);
-        if (entry == null)
-          entry = {"logins": 1};
-        else
-        entry.logins++;
+    let chartDataMap = new Map<string, MainChartData>();
```

Comment:
Well, in case the separation of logs into workspace would be needed, it will be there (bigger dropdown chart, workspace filter for example). so it is simply for futureproofing and nicer look overall (easier to work with sorted data)